SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc

```diff
@@ -439,24 +440,26 @@ void HcalDigitizer::accumulateCaloHits(edm::Handle<std::vector<PCaloHit> > const
     std::vector<PCaloHit> hcalHits;
     hcalHits.reserve(hcalHitsOrig.size());
 
-    for ( unsigned int i=0; i< hcalHitsOrig.size(); i++) {
-      DetId id(hcalHitsOrig[i].id());
-      HcalDetId hid(id);
-      
-      //      if ( id.subdetId()==2 && hid.ietaAbs()<16 ) edm::LogError("HcalDigitizer") << "bad hcal id found in digitizer. Skipping " << id.rawId() << std::endl;
-      if ( !htopoP->validHcal(hid) ) edm::LogError("HcalDigitizer") << "bad hcal id found in digitizer. Skipping " << id.rawId() << std::endl;
-      else
-	hcalHits.push_back(hcalHitsOrig[i]);
-    }
     //evaluate darkening before relabeling
     if(m_HEDarkening || m_HFRecalibration){
-      darkening(hcalHits);
+      darkening(hcalHitsOrig);
```

Comment:
Both darkening and relableller likes the unsigned in in detId packed by hcaltestnumberingscheme. We may not have a valid function which validate both type of packing. We may have to provide one if you want. The code as it is now works fine for Run1 and 2 with protection and PhaseI and II without protection.
________________________________
From: Kevin Pedro [notifications@github.com]
Sent: 05 November 2015 16:47
To: cms-sw/cmssw
Cc: Sunanda Banerjee
Subject: Re: [cmssw] HGC ToT Digitization Implementation (and corresponding RECO updates) (#11880)


In SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc<https://github.com/cms-sw/cmssw/pull/11880#discussion_r44026765>:

>      //evaluate darkening before relabeling
>      if(m_HEDarkening || m_HFRecalibration){
> -      darkening(hcalHits);
> +      darkening(hcalHitsOrig);


The darkening() function expects an HcalTestNumbering ID:
https://github.com/lgray/cmssw/blob/hgc_fasttime_fromtdconset_76X/SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc#L717-L722

I'm not sure what HcalTestNumbering::unpackHcalIndex() will do with an invalid ID, but again, darkening will only be enabled in the appropriate situations. It is not a commonly used feature.

—
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/11880/files#r44026765>.